src/Interpreters/Context.cpp

```diff
@@ -3646,6 +3647,8 @@ void Context::setDefaultProfiles(const Poco::Util::AbstractConfiguration & confi
     setCurrentProfile(shared->system_profile_name);
 
     applySettingsQuirks(settings, &Poco::Logger::get("SettingsQuirks"));
+    if (shared->application_type == ApplicationType::LOCAL)
```

Comment:
We can put this into Local.cpp, and encapsulate the method applySettingsOverridesForLocal there as well.

Reply:
>We can put this into Local.cpp

Yes, this is better. By some reason I thought that it will not be possible to apply those settings for buffer_context, but I was wrong, everything is OK in this front (even though it is not that important).

>and encapsulate the method applySettingsOverridesForLocal there as well

I thought that keeping it near to the SettingsQuirks.cpp was better, but I guess this is my subjective opinion.

Fixed.